src/components/experiment-tracking/run-dataset/run-dataset.js

```diff
@@ -139,17 +172,64 @@ function buildDatasetDataMarkup(
         >
           {datasetKey}
         </span>
-        {runDataWithPin.map((data, index) => (
-          <span
-            className={classnames('details-dataset__value', {
-              'details-dataset__value--single': isSingleRun,
-            })}
-            key={data.runId + index}
-          >
-            {sanitizeValue(data.value)}
-            {enableShowChanges && <PinArrowIcon icon={data.pinIcon} />}
-          </span>
-        ))}
+        {isTracking &&
+          runDataWithPin.map((data, index) => (
+            <span
+              className={classnames('details-dataset__value', {
+                'details-dataset__value--single': isSingleRun,
+              })}
+              key={data.runId + index}
+            >
+              {sanitizeValue(data.value)}
+              {enableShowChanges && <PinArrowIcon icon={data.pinIcon} />}
+            </span>
+          ))}
+        {isPlotly &&
+          runDataWithPin.map((data, index) => {
+            return (
+              <>
+                <span
+                  className={classnames('details-dataset__value', {
+                    'details-dataset__value--single': isSingleRun,
+                  })}
+                  key={data.runId + index}
```

Comment:
`key` should be added to the parent element in this case. If it's here, and the parent or sibling changes, React might not be able to keep track.